index.bs

```diff
@@ -555,6 +701,223 @@ Report Long Animation Frames {#loaf-processing-model}
+    1. If |document| is not [=fully active=] or	{{Document/hidden}}, then return.
+    1. Let |frameTimingInfo| be |document|'s [=relevant frame timing info=].
+    1. Let |scriptTimingInfo| be |frameTimingInfo|'s [=frame timing info/pending script=].
+    1. Set |frameTimingInfo|'s [=frame timing info/pending script=] to null.
+    1. If |scriptTimingInfo| is null, then return.
+    1. Set |scriptTimingInfo|'s [=script timing info/end time=] to the [=unsafe shared current time=].
+    1. If |script| is a [=classic script=] whose [=classic script/muted errors=] is true, then:
+        1. set |scriptTimingInfo|'s [=script timing info/source url=] to the empty string.
+        1. set |scriptTimingInfo|'s [=script timing info/source character position=] to -1.
+        1. set |scriptTimingInfo|'s [=script timing info/source function name=] to the empty string.
+    1. If the [=duration=] between |scriptTimingInfo|'s [=script timing info/start time=] and |scriptTimingInfo|'s [=script timing info/end time=] is greater than 5 milliseconds, then
+        [=list/append=] |scriptTimingInfo| to |frameTimingInfo|'s [=frame timing info/scripts=].
+</div>
+
+<div algorithm="Applying source location">
+    To <dfn>apply source location</dfn> to a [=script timing info=] |scriptTimingInfo| given a [=callback function=] or {{Function}} |callback|:
+    1. The user agent may set |scriptTimingInfo|'s [=script timing info/source url=] to the source URL of the script where |callback| was defined.
+    1. The user agent may set |scriptTimingInfo|'s [=script timing info/source function name=] to the function name of |callback|.
+    1. The user agent may set |scriptTimingInfo|'s [=script timing info/source character position=] to the character position where |callback| was defined.
+</div>
+
+Additions to existing standards {#other-standards}
+==================================================
+
+Monkey-patches to the WebIDL standard {#webidl-monkey-patches}
+----------------------------------------------------------
+<div algorithm="Monkey patches to promise handling">
+The {{Promise}} interface has an associated string <dfn for=Promise>invoker name when created</dfn>, initially "`Promise`".
+The {{Promise}} interface has an associated string <dfn for=Promise>script url when created</dfn>, initially the empty string.
+
+Append the following steps to <a href="https://webidl.spec.whatwg.org/#a-new-promise">creating a new promise</a>, before returning the {{Promise}}:
+    1. Let |interfaceName| be a string representing the [=interface=] responsible for creating this promise.
+    1. Let |attributeName| be a string representing the [=attribute=] in the interface responsible for creating this promise.
+    1. Set the created {{Promise}}'s [=Promise/script url when created=] to the [=running script=]'s [=script/base URL=].
+    1. The user-agent may set the created {{Promise}}'s [=Promise/invoker name when created=] to the last known [=concatenate|concatenation=] of
```

Comment:
"May" is strange here -- if the text is being added, why not make this imperative? Are there known circumstances where the UA cannot or should not?

Also, I'm not sure how to interpret "last-known" in this context. `interfaceName` and `attributeName` are known in the preceding steps of this algorithm.

Reply:
Yea, it's vague because I couldn't find a way to spec this normatively, and how this works might differ between implementations. So I left this a bit open.

Reply:
Added an inline issue as I don't currently have a solution for this